packages/expo-modules-core/ios/Fetch/NativeResponse.swift

```diff
@@ -0,0 +1,216 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+/**
+ A SharedRef for response.
+ */
+internal final class NativeResponse: SharedRef<ResponseSink>, ExpoURLSessionTaskDelegate {
+  private let dispatchQueue: DispatchQueue
+
+  private(set) var state: ResponseState = .intialized {
+    didSet {
+      dispatchQueue.async { [weak self] in
+        guard let self else {
+          return
+        }
+        self.stateChangeOnceListeners.removeAll { $0(self.state) == true }
+      }
+    }
+  }
+  private typealias StateChangeListener = (ResponseState) -> Bool
+  private var stateChangeOnceListeners: [StateChangeListener] = []
+
+  private(set) var responseInit: NativeResponseInit?
+  private(set) var redirected = false
+  private(set) var error: Error?
+
+  var bodyUsed: Bool {
+    return self.ref.bodyUsed
+  }
+
+  init(dispatchQueue: DispatchQueue) {
+    self.dispatchQueue = dispatchQueue
+    super.init(ResponseSink())
+  }
+
+  func startStreaming() {
+    if isInvalidState([.responseReceived]) {
+      return
+    }
+    state = .bodyStreamingStarted
+    let queuedData = self.ref.finalize()
+    emit(event: "didReceiveResponseData", arguments: queuedData)
+  }
+
+  func cancelStreaming() {
+    if isInvalidState([.bodyStreamingStarted]) {
+      return
+    }
+    state = .bodyStreamingCancelled
+  }
+
+  func emitRequestCancelled() {
```

Comment:
```suggestion
  func emitRequestCanceled() {
```

Use American English 😉